packages/wallet/src/services/HandlesTracker.ts

```diff
@@ -83,5 +128,19 @@ export const createHandlesTracker = ({ tip$, assetInfo$, handlePolicyIds, logger
       );
     }),
     distinctUntilChanged((a, b) => sameArrayItems(a, b, handleInfoEquals)),
+    map((handles) =>
+      from(handles).pipe(
+        concatMap((handle) =>
+          of(hydrateHandle(handleProvider, logger)(handle)).pipe(
+            delay(RESOLVE_HANDLE_DELAY),
+            mergeMap((x) => x)
+          )
+        ),
+        take(handles.length),
+        toArray(),
+        handles.length > 0 ? startWith(handles) : tap((x) => x)
```

Comment:
We have this for this exact purpose in `util-rxjs` package
```suggestion
        handles.length > 0 ? startWith(handles) : passthrough()
```

Reply:
Fixed here: https://github.com/input-output-hk/cardano-js-sdk/commit/2ecb53301b9c37c6727088f9fe3ac58c4d2592db.